core/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolverJob.java

```diff
@@ -275,4 +281,23 @@ public void solvingStarted(SolverScope<Solution_> solverScope) {
             }
         }
     }
+
+    /**
+     * A listener that consumes the solution from a phase only if the phase initializes the solution.
+     */
+    private final class InitializedSolutionPhaseListener extends PhaseLifecycleListenerAdapter<Solution_> {
+
+        private final ConsumerSupport<Solution_, ProblemId_> consumerSupport;
+
+        public InitializedSolutionPhaseListener(ConsumerSupport<Solution_, ProblemId_> consumerSupport) {
+            this.consumerSupport = consumerSupport;
+        }
+
+        @Override
+        public void phaseEnded(AbstractPhaseScope<Solution_> phaseScope) {
+            if (phaseScope.isInitializationPhase()) {
```

Comment:
This seems to proliferate both to the phase, and to the phase scope. Is there any chance how we can only keep it on one of the two? It feels to me like this change touches more places than it needs to; if the information is available on the phase, or if it is available on the scope, why would it also need to be available on the other?

Reply:
Okay. Let me check it.